src/panels/lovelace/cards/hui-media-control-card.ts

```diff
@@ -59,6 +59,103 @@ function getContrastRatio(
   return Math.round((contrast(rgb1, rgb2) + Number.EPSILON) * 100) / 100;
 }
 
+// For debug purposes, is being tree shaken.
+const DEBUG_COLOR = __DEV__ && false;
+
+const logColor = (
+  color: Swatch,
+  label: string = `${color.getHex()} - ${color.getPopulation()}`
+) =>
+  // tslint:disable-next-line:no-console
+  console.log(
+    `%c${label}`,
+    `color: ${color.getBodyTextColor()}; background-color: ${color.getHex()}`
+  );
+
+const customGenerator = (colors: Swatch[]) => {
+  colors.sort((colorA, colorB) => colorB.population - colorA.population);
+
+  const backgroundColor = colors[0];
+  let foregroundColor: string | undefined;
+
+  const contrastRatios = new Map<Swatch, number>();
+  const approvedContrastRatio = (color: Swatch) => {
+    if (!contrastRatios.has(color)) {
+      contrastRatios.set(
+        color,
+        getContrastRatio(backgroundColor.rgb, color.rgb)
+      );
+    }
+
+    return contrastRatios.get(color)! > CONTRAST_RATIO;
+  };
+
+  // We take each next color and find one that has better contrast.
+  for (let i = 1; i < colors.length && foregroundColor === undefined; i++) {
+    // If this color matches, score, take it.
+    if (approvedContrastRatio(colors[i])) {
+      if (DEBUG_COLOR) {
+        logColor(colors[i], "PICKED");
+      }
+      foregroundColor = colors[i].hex;
+      break;
+    }
+
+    // This color has the wrong contrast ratio, but it is the right color.
+    // Let's find similar colors that might have the right contrast ratio
+
+    const currentColor = colors[i];
+    if (DEBUG_COLOR) {
+      logColor(colors[i], "Finding similar color with better contrast");
+    }
+
+    for (let j = i + 1; j < colors.length; j++) {
```

Comment:
Little late now but Node-Vibrant has a DeltaE94 function we could probably use for this

Reply:
https://github.com/akfish/node-vibrant#deltae94lab1-number-lab2-number-number

Reply:
Played with it. It doesn't seem to be a good match. If you look at the diff scores, there are colors considered "similar" but they have a complete different hue.

![image](https://user-images.githubusercontent.com/1444314/76638609-f6ec1700-6509-11ea-9ab9-675699fbc438.png)